hack/machineset.sh

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env bash
+
+# machineset.sh - create a Windows MachineSet for Windows Machine Config Operator
+# Creates a MachineSet.yaml file and apply/delete the MachineSet if optional action is provided
+#
+# USAGE
+#    machineset.sh
+# OPTIONS
+#    $1      Action       (Optional) apply/delete the MachineSet
+# PREREQUISITES
+#    oc                   to fetch cluster info and apply/delete MachineSets on the cluster(cluster should be logged in)
+#    aws                  to fetch Windows AMI id for AWS platform (only required for clusters running on AWS)
+
+function error-exit() {
+    echo "Error: $*" >&2
+    exit 1
+}
+
+ACTION=$1
+
+# get_spec returns the template yaml common for all cloud providers
+get_spec(){
+
+local infraID=$1
+local az=$2
+
+cat <<EOF
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: $(echo $infraID)
+  name: $(echo $infraID)-windows-worker-$(echo $az)
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: $(echo $infraID)
+      machine.openshift.io/cluster-api-machineset: $(echo $infraID)-windows-worker-$(echo $az)
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: $(echo $infraID)
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: $(echo $infraID)-windows-worker-$(echo $az)
+        machine.openshift.io/os-id: Windows
+    spec:
+      metadata:
+        labels:
+          node-role.kubernetes.io/worker: ""
+EOF
+}
+
+# get_aws_ms creates a MachineSet for AWS Cloud Provider
+get_aws_ms(){
```

Comment:
Please indent the function body

Reply:
Kept the indentation at 0 because of 
```
cat <<EOF
...
EOF
```
In case, i indent, I'll have to keep the data in cat part at 0 so that it'll not result in a incorrectly indented yaml file

Reply:
Multiple solutions for this:
- You can use `echo -e` and use escape characters
- use `-EOF` with `cat
- use `sed` to fix things up after redirecting
You can experiment and figure out what works best for you.

Reply:
i'll try these out and update....

Reply:
Left the heredoc part as it is and indented the rest of body.. 
Was not able to make the indentation work as required in the heredoc